packages/modules/viewer-react/src/components/error/ErrorBoundary.tsx

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { ErrorPageProps, ErrorPageType } from "@itwin/itwinui-react";
+import { ErrorPage } from "@itwin/itwinui-react";
+import React, { Component } from "react";
+
+interface Props {
+  eventTitle?: string;
+}
+
+interface Error {
+  code?: string;
+  message?: string;
+}
+
+export class ErrorBoundary extends Component<
+  Props,
+  {
+    fallback: boolean;
+    error: Error;
+  }
+> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      fallback: false,
+      error: {},
+    };
+  }
+
+  static getDerivedStateFromError(e: Error): {
+    fallback: boolean;
+    error: Error;
+  } {
+    return {
+      fallback: true,
+      error: e,
+    };
+  }
+
+  override render(): JSX.Element {
+    if (this.state.fallback) {
+      const errorType: ErrorPageType = "generic";
+      let errorMessage: ErrorPageProps["errorMessage"] = (
```

Comment:
what is the type of this supposed to be? is this supposed to be a string or a JSX.Element?

Reply:
ok, take a look at line 55, not sure how youre assigning a string to a react node

Reply:
disregard, 
![image](https://user-images.githubusercontent.com/11051042/171261884-6b3b3edc-a9da-469d-99fb-a20ea767d010.png)

Reply:
That makes sense. When I did console.log(typeof errorMessage) it turned from an object to a string, but no errors. For consistency, I suppose I can put this.state.error.message into a React fragment though.

Reply:
I guess its fine as is. I question that default message you provide. Error.message should always be defined if an Error exists right?

Reply:
Hmm, so we don't need a default message?

Reply:
well if error.message is always defined, then we'd never have to fall back to the default message right?